lib/enumerable.rb

```diff
@@ -55,6 +55,8 @@ def my_all?(pattern = nil, &block)
         return false if e.class != String
         return false if pattern.match(e).nil?
       end
+    else
+      my_each { |e| return false if e } unless pattern.nil?
```

Comment:
Add to cover the cases when there is numeric pattern [].my_all?(3), [1,3].my_all?(3)